src/packagedcode/about.py

```diff
@@ -57,12 +57,8 @@ class AboutPackage(models.Package):
     def recognize(cls, location):
         yield parse(location)
 
-    @classmethod
-    def get_package_root(cls, manifest_resource, codebase):
-        # FIXME: this should have been already stored with the Package itself as extra_data
-        with io.open(manifest_resource.location, encoding='utf-8') as loc:
-            package_data = saneyaml.load(loc.read())
-        about_resource = package_data.get('about_resource')
+    def get_package_root(self, manifest_resource, codebase):
```

Comment:
I've changed `get_package_root()` in `AboutPackage` from a classmethod to a regular instance method because I would like to access the `extra_data` field to get the `about_resource` value. I also modified https://github.com/nexB/scancode-toolkit/blob/31e3fee011dfb7fb671e9cdeb0ab7fc23f6c7a04/src/packagedcode/plugin_package.py#L137 to instantiate the package rather than just get the package class so the `extra_data` field is accessable. However, this introduces inconsistency on how the `get_package_root` method is called across packages, where this package requires the package instance to use the method whereas other packages do not.

Reply:
Hum... I had not noticed that part. Let me review first before merging then

Reply:
This has been fixed now ... I also improved get_package_instance significantly.